pyuvdata/uvcal/uvcal.py

```diff
@@ -1264,6 +1682,374 @@ def _check_freq_spacing(self, *, raise_errors=True):
+            cat_name=cat_name,
+            cat_type=cat_type,
+            cat_lon=cat_lon,
+            cat_lat=cat_lat,
+            cat_frame=cat_frame,
+            cat_epoch=cat_epoch,
+            cat_times=cat_times,
+            cat_pm_ra=cat_pm_ra,
+            cat_pm_dec=cat_pm_dec,
+            cat_dist=cat_dist,
+            cat_vrad=cat_vrad,
+            info_source=info_source,
+        )
+
+        # We want to create a unique ID for each source, for use in indexing arrays.
+        # The logic below ensures that we pick the lowest positive integer that is
+        # not currently being used by another source
+        if cat_id is None or not force_update:
+            cat_id = uvutils.generate_new_phase_center_id(
+                phase_center_catalog=self.phase_center_catalog, cat_id=cat_id
+            )
+
+        if self.phase_center_catalog is None:
+            # Initialize an empty dict to plug entries into
+            self.phase_center_catalog = {}
+        else:
+            # Let's warn if this entry has the same name as an existing one
+            temp_id, cat_diffs = uvutils.look_in_catalog(
+                self.phase_center_catalog, phase_dict=cat_entry
+            )
+
+            # If the source does have the same name, check to see if all the
+            # attributes match. If so, no problem, go about your business
+            if temp_id is not None:
+                if cat_diffs == 0:
+                    # Everything matches, return the catalog ID of the matching entry
+                    return temp_id
+                warnings.warn(
+                    f"The provided name {cat_name} is already used but has different "
+                    "parameters. Adding another entry with the same name but a "
+                    "different ID and parameters."
+                )
+
+        # If source is unique, begin creating a dictionary for it
+        self.phase_center_catalog[cat_id] = cat_entry
+        self.Nphase = len(self.phase_center_catalog)
+
+        return cat_id
+
+    def _remove_phase_center(self, defunct_id):
+        """
+        Remove an entry from the internal object/source catalog.
+
+        Removes an entry from the attribute `phase_center_catalog`.
+
+        Parameters
+        ----------
+        defunct_id : int
+            Catalog ID of the source to be removed
+
+        Raises
+        ------
+        IndexError
+            If the name provided is not found as a key in `phase_center_catalog`
+        """
+        if defunct_id not in self.phase_center_catalog:
+            raise IndexError("No source by that ID contained in the catalog.")
+
+        del self.phase_center_catalog[defunct_id]
+        self.Nphase = len(self.phase_center_catalog)
+
+    def _clear_unused_phase_centers(self):
+        """
+        Remove objects dictionaries and names that are no longer in use.
+
+        Goes through the `phase_center_catalog` attribute in of a UVData object and
```

Comment:
UVData -> UVCal

Reply:
Fixed in 766e32646768181b70ddd8d2e576d90ca57c254c.